File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.c
#include <stdio.h>
#include <math.h>

void hanoi_tower(int n, int from, int temp, int to);


int main(void) {
	int n;
	scanf("%d", &n);
	int cnt = pow(2, n) - 1;
	printf("%d\n", cnt);
	hanoi_tower(n, 1, 2, 3);

	
	return 0; 
}

void hanoi_tower(int n, int from, int temp, int to) {
	if (n == 1) {
		printf("%d %d\n", from, to);
	}
	else {
		hanoi_tower(n - 1, from, to, temp);
		printf("%d %d\n", from, to);
		hanoi_tower(n - 1, temp, from, to);
	}

}

File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.py
def hanoi_tower(n, start, middle, end):
    if(n == 1):
        print(start, end)
    else:
        hanoi_tower(n-1, start, end, middle)
        print(start, end)
        hanoi_tower(n - 1, middle, start, end)
    

n = int(input())
cnt = int(2 ** n - 1)
print(cnt)
hanoi_tower(n, 1, 2, 3)